CONTRIBUTING.md

```diff
@@ -1,112 +1,80 @@
 
-#### Reviews & Approvals
+```bash
+ulimit -Sn 3000
+```
 
-All pull requests must be reviewed and approved by at least one MeiliSearch team
-member.
+## Git Guidelines
 
-#### Merge Style
-
-All pull requests are squashed and merged. We generally discourage large pull
-requests that are over 300-500 lines of diff. If you would like to propose
-a change that is larger we suggest coming onto our chat channel and
-discuss it with one of our engineers. This way we can talk through the
-solution and discuss if a change that large is even needed! This overall
-will produce a quicker response to the change and likely produce code that
-aligns better with our process.
-
-## Development
+### Git Branches
 
-### Setup
+All changes must be made in a branch and submitted as PR.
 
-See the [MeiliSearch Docs](https://docs.meilisearch.com/reference/features/installation.html) for how to set up a development environment.
+We do not enforce any branch naming style, but please use something descriptive of your changes.
 
-### Benchmarking & Profiling
+### Git Commits
 
-We do not yet do any benchmarking, nor have we formalised our profiling. If you'd like to work on this please get in touch!
+As minimal requirements, your commit message should:
+- be capitalized
+- not finish by a dot or any other punctuation character (!,?)
+- start with a verb so that we can read your commit message this way: "This commit will ...", where "..." is the commit message.
+  e.g.: "Fix the home page button" or "Add more tests for create_index method"
 
-## Humans
+We don't follow any other convention, but if you want to use one, we recommend [this one](https://chris.beams.io/posts/git-commit/).
 
-After making your change, you'll want to prepare it for MeiliSearch users (mostly humans). This usually entails updating documentation and announcing your feature.
+### Github Pull Requests
 
-### Documentation
+Some notes on GitHub PRs:
 
-Documentation is very important to MeiliSearch. All contributions that
-alter user-facing behavior MUST include documentation changes. Please see
-[GitHub.com/meilisearch/documentation](https://github.com/meilisearch/documentation) for more info.
+- All PRs must be reviewed and approved by at least one maintainer.
+- The PR title should be accurate and descriptive of the changes.
+- [Convert your PR as a draft](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-stage-of-a-pull-request) if your changes are a work in progress: no one will review it until you pass your PR as ready for review.<br>
+  The draft PR can be very useful if you want to show that you are working on something and make your work visible.
+- The branch related to the PR must be **up-to-date with `main`** before merging. Fortunately, this project [Bors](https://github.com/bors-ng/bors-ng) to automatically enforce this requirement without the PR author having to do it manually..
```

Comment:
```suggestion
- The branch related to the PR must be **up-to-date with `main`** before merging. Fortunately, this project uses [Bors](https://github.com/bors-ng/bors-ng) to automatically enforce this requirement without the PR author having to rebase manually, most of the time it works.
```

Reply:
I would rather not use the `most of the time it works` 😂